ztools/ztls/common.go

```diff
@@ -949,3 +982,157 @@ type ClientHelloInfo struct {
 	// connection to fail.
 	Conn net.Conn
 }
+
+func (info *ClientHelloInfo) MarshalJSON() ([]byte, error) {
+	aux := struct {
+		CipherSuites      []CipherSuite     `json:"cipher_suites"`
+		ServerName        string            `json:"server_name,omitempty"`
+		SupportedCurves   []CurveID         `json:"supported_curves,omitempty"`
+		SupportedPoints   []PointFormat     `json:"supported_point_formats,omitempty"`
+		SignatureSchemes  []SignatureScheme `json:"signature_schemes,omitempty"`
+		SupportedProtos   []string          `json:"supported_protocols,omitempty"`
+		SupportedVersions []TLSVersion      `json:"supported_versions,omitempty"`
+		LocalAddr         string            `json:"local_address,omitempty"`
+		RemoteAddr        string            `json:"remote_address,omitempty"`
+	}{
+		ServerName:       info.ServerName,
+		SupportedCurves:  info.SupportedCurves,
+		SignatureSchemes: info.SignatureSchemes,
+		SupportedProtos:  info.SupportedProtos,
+	}
+
+	aux.CipherSuites = make([]CipherSuite, len(info.CipherSuites))
+	for i, cipher := range info.CipherSuites {
+		aux.CipherSuites[i] = CipherSuite(cipher)
+	}
+
+	aux.SupportedPoints = make([]PointFormat, len(info.SupportedPoints))
+	for i, format := range info.SupportedPoints {
+		aux.SupportedPoints[i] = PointFormat(format)
+	}
+
+	aux.SupportedVersions = make([]TLSVersion, len(info.SupportedVersions))
+	for i, version := range info.SupportedVersions {
+		aux.SupportedVersions[i] = TLSVersion(version)
+	}
+
+	aux.LocalAddr = fmt.Sprintf("%s+%s", info.Conn.LocalAddr().String(), info.Conn.LocalAddr().Network())
```

Comment:
No this is a trimmed down version exposed by Golang 1.8 and not the entire Client Hello.